internal/file/filedetails.go

```diff
@@ -1,23 +1,25 @@
 package file
 
 import (
+	"fmt"
 	"os"
 )
 
-type filedetails struct {
-	name string
-	info os.FileInfo
-	hash Hash // Used to check if the file has changed on disk since loaded.
+type Filedetails struct {
+	Name string
```

Comment:
This is temporary because you need to reach into this? Because once `File` et al. are moved into `internal/file` these should be internal details of `file.Details` right? And hence lower case? Add a `TODO`.

Reply:
Yes it is just temporary, eventually the logic will be changed but at present, there are still things accessing the name.